src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -119,28 +123,54 @@ public static bool TryRegisterSocket(IntPtr socketHandle, SocketAsyncContext con
 
         private bool TryRegisterCore(IntPtr socketHandle, SocketAsyncContext context, out Interop.Error error)
         {
-            bool added = _handleToContextMap.TryAdd(socketHandle, new SocketAsyncContextWrapper(context));
-            if (!added)
+            Debug.Assert(context.GlobalContextIndex == -1);
+
+            lock (s_registeredContextsFreeList)
             {
-                // Using public SafeSocketHandle(IntPtr) a user can add the same handle
-                // from a different Socket instance.
-                throw new InvalidOperationException(SR.net_sockets_handle_already_used);
```

Comment:
There are weird reports of the deleted exception popping up, see #64305 and #56750. This is caused by buggy app or library code according to our current understanding, however the change seems to make duplicate registrations virtually undetectable making potential issues even harder to diagnose.

Is this something we are willing to accept or should we consider some mitigation?

Reply:
I'd defer to you on whether it's valuable to keep the detection around, it's not a ton of logic to keep doing so if we care (extra dictionary on the side like we had before). My impression was that we were doing more hand holding than I'd expect for working with handles (it was "free" with the previous appraoch), but I haven't been following those discussions.

Reply:
I see now that `Private.InternalDiagnostics` actually logs handle alloc/free leaving us a chance to diagnose such errors if they occur again, so I'm fine with the change as-is.

FYI: problems came from users (even library authors) leaving copypasta from [our dispose pattern docs](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern), see https://github.com/dotnet/docs/issues/44201.